source/calculate-retry-delay.ts

```diff
@@ -1,40 +1,47 @@
 import is from '@sindresorhus/is';
 import {HTTPError, ParseError, MaxRedirectsError, GotError} from './errors';
-import {RetryFunction} from './utils/types';
+import {RetryFunction, RetryObject} from './utils/types';
 
 const retryAfterStatusCodes: ReadonlySet<number> = new Set([413, 429, 503]);
 
+const isErrorWithResponse = (error: RetryObject['error']): error is HTTPError | ParseError | MaxRedirectsError => (
+	error instanceof HTTPError || error instanceof ParseError || error instanceof MaxRedirectsError
+);
+
 const calculateRetryDelay: RetryFunction = ({attemptCount, retryOptions, error}) => {
 	if (attemptCount > retryOptions.limit) {
 		return 0;
 	}
 
-	const hasMethod = retryOptions.methods.includes((error as GotError).options.method);
-	const hasErrorCode = Reflect.has(error, 'code') && retryOptions.errorCodes.includes((error as GotError).code);
-	const hasStatusCode = Reflect.has(error, 'response') && retryOptions.statusCodes.includes((error as HTTPError | ParseError | MaxRedirectsError).response.statusCode);
-	if (!hasMethod || (!hasErrorCode && !hasStatusCode)) {
-		return 0;
-	}
-
-	// TODO: This type coercion is not entirely correct as it makes `response` a guaranteed property, when it's in fact not.
-	const {response} = error as HTTPError | ParseError | MaxRedirectsError;
-	if (response && Reflect.has(response.headers, 'retry-after') && retryAfterStatusCodes.has(response.statusCode)) {
-		let after = Number(response.headers['retry-after']);
-		if (is.nan(after)) {
-			after = Date.parse(response.headers['retry-after']) - Date.now();
-		} else {
-			after *= 1000;
-		}
-
-		if (after > retryOptions.maxRetryAfter) {
+	if (error instanceof GotError) {
```

Comment:
Isn't this always true?

Reply:
I am not sure - in tests I see a check on user-defined custom error codes, so it might not be?

Reply:
But it's still GotError, isn't it? It's created in `onError` function IIRC.

Reply:
I checked the source code - it seems that there is only one place calling `calculateRetryDelay`: `emitter.retry`. It is being called in two places, with the possible error choices `TimeoutError`, `HTTPError` and `RequestError`. I think you are right - I will update to reflect this in the code.